### PR TITLE
AM-1226(gateway): wip: enroll step updated

### PR DESCRIPTION
**Pull request by ashraf706**
PR to update data model

### CODE REVIEW

File: gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java [hunk 1]
```diff
@@ -27,12 +27,50 @@
...
     public ChallengeSettings(ChallengeSettings other) {
         this.active = other.active;
-        this.forceEnrollment = other.forceEnrollment;
+        this.forceChallenge = other.forceChallenge;
+        this.challengeRule = other.challengeRule;
+        this.type = other.type;
+    }
+
+
+    public boolean isActive() {
```
Comment (mstasgravitee) [inline gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java:45]: setters and getters can be removed

File: gravitee-am-model/src/main/java/io/gravitee/am/model/EnrollSettings.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.model;
+
+/**
+ * @author Ashraful HASAN (ashraful.hasan at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class EnrollSettings {
```
Comment (mstasgravitee) [inline gravitee-am-model/src/main/java/io/gravitee/am/model/EnrollSettings.java:22]: @Getter
@Setter
@NoArgsConstructor

File: gravitee-am-model/src/main/java/io/gravitee/am/model/EnrollmentSettings.java [hunk 3]
```diff
@@ -27,15 +27,29 @@
 @Setter
 @NoArgsConstructor
```
Comment (mstasgravitee) [inline gravitee-am-model/src/main/java/io/gravitee/am/model/EnrollmentSettings.java:28]: we can remove annotation or setters / getters

File: gravitee-am-model/src/main/java/io/gravitee/am/model/MFASettings.java [hunk 4]
```diff
@@ -64,5 +67,73 @@ public MFASettings(MFASettings other) {
                 .filter(Objects::nonNull)
                 .map(StepUpAuthenticationSettings::new)
                 .orElse(new StepUpAuthenticationSettings());
+        this.enroll = ofNullable(other.enroll)
+                .filter(Objects::nonNull)
+                .map(EnrollSettings::new)
+                .orElse(new EnrollSettings());
+    }
+
+    public String getLoginRule() {
```
Comment (mstasgravitee) [inline gravitee-am-model/src/main/java/io/gravitee/am/model/MFASettings.java:76]: we can remove annotation or setters / getters

File: gravitee-am-model/src/main/java/io/gravitee/am/model/MFASettings.java [hunk 5]
```diff
@@ -39,8 +39,11 @@ public class MFASettings {
     private String skipStepUpAuthentication;
     private String adaptiveAuthenticationRule;
```
Comment (mstasgravitee) [inline gravitee-am-model/src/main/java/io/gravitee/am/model/MFASettings.java:40]: @Deprecated

File: gravitee-am-model/src/main/java/io/gravitee/am/model/MFASettings.java [hunk 6]
```diff
@@ -39,8 +39,11 @@ public class MFASettings {
     private String skipStepUpAuthentication;
```
Comment (mstasgravitee) [inline gravitee-am-model/src/main/java/io/gravitee/am/model/MFASettings.java:39]: we can remove it or move it to new enrollment settings

File: gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java [hunk 7]
```diff
@@ -27,12 +27,50 @@
 @Setter
 @NoArgsConstructor
 public class ChallengeSettings {
-    private Boolean active;
-    private Boolean forceEnrollment;
     private String option;
```
Comment (mstasgravitee) [inline gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java:30]: to remove

File: gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java (hunk 1 above)
Comment (ashraf706) [reply gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java:45]: done

File: gravitee-am-model/src/main/java/io/gravitee/am/model/EnrollSettings.java (hunk 2 above)
Comment (ashraf706) [reply gravitee-am-model/src/main/java/io/gravitee/am/model/EnrollSettings.java:22]: done

File: gravitee-am-model/src/main/java/io/gravitee/am/model/EnrollmentSettings.java (hunk 3 above)
Comment (ashraf706) [reply gravitee-am-model/src/main/java/io/gravitee/am/model/EnrollmentSettings.java:28]: setters/getters removed

File: gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java (hunk 7 above)
Comment (ashraf706) [reply gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java:30]: done

File: gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java [hunk 8]
```diff
@@ -27,12 +27,50 @@
...
+    private boolean active;
+    private boolean forceChallenge;
```
[the remark this answers is not in the document]
Comment (ashraf706) [reply gravitee-am-model/src/main/java/io/gravitee/am/model/ChallengeSettings.java]: removed

File: gravitee-am-model/src/main/java/io/gravitee/am/model/MFASettings.java (hunk 6 above)
Comment (ashraf706) [reply gravitee-am-model/src/main/java/io/gravitee/am/model/MFASettings.java]: removed for now

**Review by mstasgravitee — APPROVED** [review]

**Pull request merged**